### PR TITLE
Move index to index-v0.11.0.db (new format) and centralize location config

### PR DESCRIPTION
**Pull request by calmh**
At some future time, should have a routine to clean up various oldcrap that we'll have accumulated in the user's config dir. This will result in a full rescan, but I don't see an easy way around that since we need to rewrite the format of all entries in the index anyway.

### CODE REVIEW

File: cmd/syncthing/main.go [hunk 1]
```diff
@@ -376,13 +372,12 @@ func syncthingMain() {
 		runtime.GOMAXPROCS(runtime.NumCPU())
 	}
 
-	events.Default.Log(events.Starting, map[string]string{"home": confDir})
+	events.Default.Log(events.Starting, locations)
```
Comment (AudriusButkevicius) [inline cmd/syncthing/main.go]: This is an API change.

**Comment by AudriusButkevicius** [comment]
Looks good.

The only other suggestion i have is to replace v2 with `index-v0.11` or something, as this way it has some more useful context.

File: cmd/syncthing/main.go (hunk 1 above)
Comment (calmh) [inline cmd/syncthing/main.go]: Yep, true. I'll add it to the REST docs. I think this is safe enough as we're anyway in breaking-the-API mode at the moment.

**Comment by calmh** [comment]
> index-v0.11

Good idea. How about `index.v0.11+` so it doesn't look out of place when we do v0.12? Or is the `+` confusing by itself? It seems to be allowed on all OS:es at least.

File: cmd/syncthing/main.go (hunk 1 above)
Comment (AudriusButkevicius) [inline cmd/syncthing/main.go]: Agree. But we should make a note that this is changing.

Comment (calmh) [inline cmd/syncthing/main.go]: Actually no, I'll keep this as it is (`map[string]string{"home": baseDir["config"]}`) instead, there's no reason to change it; it was just a reflex as I was cleaning up some other uses of `confDir`.

**Comment by dcarosone** [comment]
Or migrate on upgrade to match the running version every time, by copy if
no other migration changes are needed. Makes it clearer when old files are
no longer needed.

**Comment by calmh** [comment]
Might be nice to be able to downgrade though, if v0.12 turns out to be a clusterfuck. In that case not having renamed the index (if there were not format changes) would be a win.

**Comment by AudriusButkevicius** [comment]
I'd be in favor of explicit v0.11.0 or something, just incase we have multiple versions in a patch version.
Also, who will recycle older versions though?

I think we should glob('index*') and zap whichever one is lower than our version, as these things can take up quite some space.

**Comment by calmh** [comment]
Okay. New policy is the index is named after the version that introduces the format. We should for sure clear out old indexes, although probably some time after they've last been updated, so we don't immediately nuke things to prevent downgrades.

**Comment by AudriusButkevicius** [comment]
@st-jenkins run integration tests please

**Comment by calmh** [comment]
Hmm.

**Comment by calmh** [comment]
@st-jenkins run integration tests please

**Comment by calmh** [comment]
Various tests broken by the tests not deleting the new index dir when they should, fixed now, will see how that works...

**Comment by calmh** [comment]
@st-jenkins run integration tests please

**Comment by calmh** [comment]
There's still brokenness here...

```
[I6KAH] 11:36:05 INFO: Skipping folder default scan due to folder error: Folder marker missing
```

forever on #.1 and #.2, and

```
...
[373HS] 11:35:53 INFO: Puller (folder "default", file "0/05/057ef8d9aecca3f5459fb9e414718c03"): pull: no available source device
[373HS] 11:35:53 INFO: Puller (folder "default", file "0/03/03305a3b36d92d22a714f2a8021673bf"): pull: no available source device
[373HS] 11:35:53 INFO: Puller (folder "default", file "0/08/086d78f905f43e3221587bebdb467ffb"): pull: no available source device
...
```

forever on #.3.

**Comment by AudriusButkevicius** [comment]
Which set of tests is that?

**Comment by calmh** [comment]
http://build.syncthing.net/job/syncthing-pr-integration/52/ with logs https://nym.se/t/logs.tar.gz since they're probably gone from the workspace soon.

**Comment by AudriusButkevicius** [comment]
Just a repetition of what I said on IRC.
This forks before the marker business got merged in, hence I cannot even understand how these logs could be present in this run.

**Comment by calmh** [comment]
I rebased it on top of latest master. Going to look at it later, I suspect it's just the tests that are not fully up to speed somewhere.

Although the "no source available" repeating forever is an interesting reproduction of that issue...

**Comment by AudriusButkevicius** [comment]
I don't think it's forever though as we just have a lot of files in the queue.

I think the integration tests are failing even with the old mechanism before the marker stuff merge.

**Comment by calmh** [comment]
@st-jenkins run integration tests, and make sure they pass this time!

**Comment by AudriusButkevicius** [comment]
So what's changed versus the old version?
Can't see anything major that would suddenly solve all the problems?

**Comment by calmh** [comment]
The tests, mostly. The old tests cleaned out `h1/index` and so on. I changed that to `h1/index*` but `RemoveAll` didn't actually understand wild cards. With an old index present but no folder marker, the folders failed to start, hence fail whale all over the place.

**Pull request merged**